### ISSUE TITLE
Expired paymaster did not return error code as expected 

### ISSUE THREAD

**Issue by 108356037** [stackup-wallet/stackup-bundler#129]
Hi there, I encountered a bug like behavior when sending expired paymaster data to bundler. I implemented my own version of paymaster contract. As for the function `_validatePaymasterUserOp `, it looks something like this at the end:

```javascript
function _validatePaymasterUserOp(
      UserOperation calldata userOp,
      bytes32 /*userOpHash*/,
      uint256 maxCost
  ) internal override returns (bytes memory context, uint256 validationData) {
 
      ...

      try this._validatePriceFeed(priceFeed, signature) {
          // if success, returns (context,validationData) for further stage
          return (
              abi.encode(
                  userOp.getSender(),
                  priceFeed.token,
                  userOp.maxFeePerGas,
                  userOp.maxPriorityFeePerGas,
                  maxTokenCost,
                  maxCost
              ),
              packValidationData(
                  false,
                  priceFeed.timestamp + 2 minutes,
                  priceFeed.timestamp
              )
          );
      } catch {
          // if fail, returns (context,validationData) too,
          // but context would be empty (we don't need it) and validationData would return sigFailed
          return ("", SIG_VALIDATION_FAILED);
      }
  }
```

We pass on valid time range using the timestamp of price feed from a verified off-chain oracle. According to EIP 4337 spec and entrypoint code, this should revert the userOp if out of time range. In fact, if I send userOp with expired paymaster data to a self-host Stackup bundler, I do see logs like this below:
![Screen Shot 2023-03-29 at 4 05 57 PM](https://user-images.githubusercontent.com/65017949/228468165-d1b02183-152d-4433-bb3a-b9a447bcdb03.png)

However, on the client side, I didn't get this error response. I got the userOpHash as response as if it was a valid userOp.
My guess is it's a bug on the bundler side, but maybe I missed some details and implemented the paymaster incorrectly. (The paymaster does work if the time range is valid tho.)

Please let me know your thought on this, thank you! 😊

**Comment by hazim-j** [comment]
Nice catch! This is definitely a missing edge case for both the bundler and the [bundler test suite](https://github.com/eth-infinitism/bundler-spec-tests).

We'll need to add a test case and patch to check the simulation results for an `out of time-range` error and return a `-32503`.

We recently did a similar fix for invalid signature in #120 and this would likely be a similar approach.

**Comment by 108356037** [comment]
Got it, will wait for the code patch and test it out. Thank you!
